packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -264,6 +473,298 @@ const Dashboard = () => {
   - The navigation bar will have `role=navigation` to indicate that it is a navigation landmark.
   - TopNav sub components will only compose other components and will not have any ARIA roles or attributes.
 
+
+### Alternatives considered for new TabNav API
+
+<details>
+
+<summary>Go inside rabbit hole</summary>
+
+
+### The Problem
+
+The new interaction that we have gotten for top nav is a bit more trickier to implement than it seems while ensuring flexibility of the component. 
+
+For example, we want users to: 
+
+1. Be able to add a “More” dropdown with custom items (It’s just a Menu)
+2. When the screen size reduces, automatically inject extra items into “More” menu
+3. Have the flexibility to customize the what’s inside “More” menu
+
+**Our current API looks like this:**
+
+```jsx
+<TabNav>
+  <TabNavItemLink icon={HomeIcon} accessibilityLabel="Home" href="/home" />
+  <TabNavItemLink title="Payroll" href="/payroll" />
+  <TabNavItemLink title="Payments" href="/payments" />
+  <Menu>
+    <TabNavItemLink title="More" href="/more" />
+    <MenuOverlay>
+      {overflowingItems.map((item) => {
+        // ??
+      })}
+      <MenuItem>
+        <ExploreItem title="Magic Checkout" description="Some product description" />
+      </MenuItem>
+      <MenuItem>
+        <ExploreItem title="RazorX" description="Some product description" />
+      </MenuItem>
+    </MenuOverlay>
+  </Menu>
+</TabNav>
+```
+
+**This API has few issues:**
+
+1. While it lets you add the “More” Menu and choose initial ordering of Magic Checkout & RazorX, when screen size reduces there is no straightforward or intuitive way to move the overflowing items into the “More” Menu
+2. Even if we could somehow magically get the overflowingItems what data will item contain that will enable us to render the <ExploreItem />? We need title, description, icon etc data for rendering which the <TabNavItemLink> may not even need or be provided with since we don’t need to render description or icon. 
+
+----
+
+
+> NOTE: If you want to go into the rabbit hole you can read the alternatives below.
+
+### Alternative APIs
+
+**Table Like Data API (Accepted approach):**
+
+A data driven API where we pass the data for the items and use render props to render the items and the overflowingItems.
+
+**Pros:**
+1. Users have easy access to the overflowingItems via props
+2. Intuitive and faimilar API, similar to Table API
+3. Users have the flexibility to render TabNavItemLinks or any other component.
```

Comment:
> why do we want to provide users with the flexibility to render anything? I think it is safe to assume that TabNav will always have a TabNavItem and a Menu right?

It's about what's inside the Menu. We don't know. 
And we had a design discussion also that even pingal is not sure what might change in future. 
The layout could change (eg: two column layout inside Menu or a product specific column layout, like 1 column for payments, 1 column for payroll), menu's header/footer etc can change. 

We don't want to keep a hard dependency here because otherwise there will be a lot of blockers on product side if something changes and will introduce breaking change. 

Having this flexible means product can decide what/how and where to put things.